citest/json_predicate/path_predicate_result.py

```diff
@@ -230,12 +230,15 @@ def export_to_json_snapshot(self, snapshot, entity):
     snapshot.edge_builder.make_mechanism(entity, 'Predicate', self.__pred)
     snapshot.edge_builder.make_output(
         entity, 'Path Values', self.__path_values)
-    snapshot.edge_builder.make_output(
-        entity, 'Pruned Paths', self.__path_failures)
-    snapshot.edge_builder.make_output(
-        entity, 'Rejected Values', self.__invalid_candidates)
-    snapshot.edge_builder.make_output(
-        entity, 'Value Justification', self.__valid_candidates)
+    if self.__path_failures:
+      snapshot.edge_builder.make_output(
+          entity, 'Pruned Paths', self.__path_failures)
+    if self.__invalid_candidates:
+        snapshot.edge_builder.make_output(
+          entity, 'Rejected Values', self.__invalid_candidates)
+    if self.__valid_candidates:
+      snapshot.edge_builder.make_output(
+          entity, 'Value Justification', self.__valid_candidates)
```

Comment:
Added comment (and changed to Value Justifications).

FYI the snapshot breaks out (duplicates) just the values because that's commonly what you want to see. However the snapshot provides the full traceability of where those values came from and why they were chosen. That is the justification. The "Rejected Values" are the same thing but for the values that didnt match the criteria. To complete the picture "Pruned Paths" are where a value could not be found at all (e.g. point out where an internal attribute might not exist because it had a different name, was collapsed, or had another intermediate node).